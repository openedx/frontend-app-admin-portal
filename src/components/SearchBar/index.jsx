import React from 'react';
import PropTypes from 'prop-types';
import { SearchField } from '@openedx/paragon';

const SearchBar = props => (
  <SearchField
    onSubmit={query => props.onSearch(query)}
    {...props}
  />
);

SearchBar.propTypes = {
  onSearch: PropTypes.func.isRequired,
};

export default SearchBar;
