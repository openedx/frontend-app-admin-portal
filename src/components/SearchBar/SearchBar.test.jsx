import React from 'react';
import renderer from 'react-test-renderer';
import { mount } from 'enzyme';

import SearchBar from './index';

describe('<SearchBar />', () => {
  let wrapper;

  it('renders correctly', () => {
    const tree = renderer
      .create((
        <SearchBar
          onSearch={() => {}}
        />
      ))
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('calls onSearch callback handler', () => {
    const mockOnSearchCallback = jest.fn();
    wrapper = mount((
      <SearchBar
        onSearch={mockOnSearchCallback}
      />
    ));
    wrapper.find('input[type="text"]').simulate('change', { target: { value: 'foobar' } });
    wrapper.find('form').simulate('submit');
    expect(mockOnSearchCallback).toHaveBeenCalledTimes(1);
  });
});
