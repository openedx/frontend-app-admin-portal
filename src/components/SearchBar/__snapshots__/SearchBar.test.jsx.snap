// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SearchBar /> renders correctly 1`] = `
<div
  className="pgn__searchfield d-flex"
>
  <form
    className="d-flex align-items-center w-100"
    onReset={[Function]}
    onSubmit={[Function]}
    role="search"
  >
    <label
      className="m-0"
      htmlFor="pgn-searchfield-input-1"
    >
      <span
        className="sr-only"
      >
        search
      </span>
    </label>
    <input
      className="form-control"
      disabled={false}
      id="pgn-searchfield-input-1"
      name="searchfield-input"
      onBlur={[Function]}
      onChange={[Function]}
      onFocus={[Function]}
      role="searchbox"
      type="text"
      value=""
    />
    <button
      className="btn"
      disabled={false}
      type="submit"
    >
      <svg
        aria-hidden="true"
        className="svg-inline--fa fa-search fa-w-16 "
        data-icon="search"
        data-prefix="fas"
        focusable="false"
        role="img"
        style={Object {}}
        viewBox="0 0 512 512"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M505 442.7L405.3 343c-4.5-4.5-10.6-7-17-7H372c27.6-35.3 44-79.7 44-128C416 93.1 322.9 0 208 0S0 93.1 0 208s93.1 208 208 208c48.3 0 92.7-16.4 128-44v16.3c0 6.4 2.5 12.5 7 17l99.7 99.7c9.4 9.4 24.6 9.4 33.9 0l28.3-28.3c9.4-9.4 9.4-24.6.1-34zM208 336c-70.7 0-128-57.2-128-128 0-70.7 57.2-128 128-128 70.7 0 128 57.2 128 128 0 70.7-57.2 128-128 128z"
          fill="currentColor"
          style={Object {}}
        />
      </svg>
      <span
        className="sr-only"
      >
        submit search
      </span>
    </button>
  </form>
</div>
`;
